layers/ip6.go

```diff
@@ -280,165 +229,191 @@ func decodeIPv6(data []byte, p gopacket.PacketBuilder) error {
 	return p.NextDecoder(ip6.NextHeader)
 }
 
-func (i *IPv6HopByHop) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
-	var bytes []byte
-	var err error
+type IPv6HeaderTLVOption interface {
```

Comment:
It appears to me that the interface must escape such a function because of IPv6AlignHeaderTLVOptions(options *[]IPv6HeaderTLVOption)? Or ipv6AlignHeaderTLVOptions(), same thing...

My comment on being idiomatic reflected more on the serialization code I wrote than anything else. I was working within the constraints of the existing API (which is excellent by the way) but this small change really ties it in for me.

My personal opinion (and it's just that) is that it's not worth trading code complexity (and possibly some functionality) for an unquantified efficiency concern. An extra malloc call seems minor, if I understand correctly. But I do understand the concern, of course.

I'm also not too cognizant on how the go compiler or the garbage collector works under the hood, altough I can make an educated guess sometimes.

Let me know, I'll work with it...

Anything you'd like to see documented? Is it just code comments or something else?